aiida_common_workflows/workflows/relax/vasp/generator.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+"""Implementation of `aiida_common_workflows.common.relax.generator.RelaxInputGenerator` for VASP."""
+from aiida import orm
+
+from ..generator import RelaxInputsGenerator, RelaxType
+from .workchain import CommonVASPRelaxWorkChain
+
+__all__ = ('VASPRelaxInputsGenerator',)
+
+
+class VASPRelaxInputsGenerator(RelaxInputsGenerator):
+    """Input generator for the `VASPRelaxWorkChain`."""
+
+    _default_protocol = 'efficiency'
+    _protocols = {'efficiency': {'description': ''}, 'precision': {'description': ''}}
+
+    _calc_types = {'relax': {'code_plugin': 'vasp.vasp', 'description': 'The code to perform the relaxation.'}}
+
+    _relax_types = {
+        RelaxType.ATOMS: 'Relax only the atomic positions while keeping the cell fixed.',
+        RelaxType.ATOMS_CELL: 'Relax both atomic positions and the cell.'
+    }
+
+    def get_builder(
+        self,
+        structure,
+        calc_engines,
+        protocol,
+        relaxation_type,
+        threshold_forces=None,
+        threshold_stress=None,
+        **kwargs
+    ):
+        """Return a process builder for the corresponding workchain class with inputs set according to the protocol.
+
+        :param structure: the structure to be relaxed
+        :param calc_engines: ...
+        :param protocol: the protocol to use when determining the workchain inputs
+        :param relaxation_type: the type of relaxation to perform, instance of `RelaxType`
+        :param threshold_forces: target threshold for the forces in eV/Å.
+        :param threshold_stress: target threshold for the stress in eV/Å^3.
+        :param kwargs: any inputs that are specific to the plugin.
+        :return: a `aiida.engine.processes.ProcessBuilder` instance ready to be submitted.
+        """
+        # pylint: disable=too-many-locals
+        code = calc_engines['relax']['code']
+        options = calc_engines['options']
+        settings = calc_engines['settings']
```

Comment:
I wasn't aware that the `calc_engine` was supposed or able to specify settings? I certainly wouldn't require it as you do here, so at the very least make it `settings = calc_engines.get('settings', {})`

Reply:
This was just WIP, so now gone. Thanks for the comment.